qiskit/result/utils.py

```diff
@@ -14,19 +14,24 @@
 
 from functools import reduce
 from re import match
+from copy import deepcopy
 
 from qiskit.exceptions import QiskitError
+from qiskit.result.result import Result
+from qiskit.result.postprocess import _bin_to_hex
 
 
-def marginal_counts(result, indices=None):
+def marginal_counts(result, indices=None, inplace=False):
```

Comment:
if I understand correctly, this changes the previous default behavior. Maybe the default should be `True`?

Reply:
I went with the semantic here since ```inplace``` suggests the original object is operated on. Since by default the result object is deep copied so it seemed like the right choice. Should I change it?

Reply:
I think the changed behavior is desired. The way it is currently almost every user ends up recreating this.

Reply:
In that case, we gonna need a release note!

Reply:
@abhik-99 you can add a release note with `reno`

Reply:
Sure @1ucian0! Would I use ```fixes``` or ```features```?

Reply:
An upgrade? It's a warning that the default behavior changed

Reply:
@1ucian0 , please have a look at commit [fe5a466](https://github.com/Qiskit/qiskit-terra/pull/4929/commits/fe5a466dd516417b886a3f45808cc1d402ddc155).